libs/dal/src/repositories/integration/integration.repository.ts

```diff
@@ -28,4 +29,35 @@ export class IntegrationRepository extends BaseRepository<IntegrationEntity> {
       'Duplicate key - One application may not have two providers of the same channel type'
     );
   }
+
+  async update(
+    query: FilterQuery<IntegrationEntity & Document>,
+    updateBody: any
+  ): Promise<{
+    matched: number;
+    modified: number;
+  }> {
+    if (query.active === true) {
+      await this.deactivatedOtherActiveChannels(query);
+    }
```

Comment:
A few considerations in bringing this sort of logic to here, which I'm not sure about.
 
the deactivation of the active channels feels like a business logic decision and like a data access point, I think it feels much appropriate to handle this in the usecase and leave the DAL to strictly handle DB writes and reads and abstract mongo query language specifics.

Imagine other usecases managing or updating other properties of the code and this line can cause some un-expected side effects.

What I suggest is using the DAL specifically for data access and leaving business rules strictly to use cases.

What do you think about this?

Reply:
I agree with you the main reason that I implemented it on the dal was that I was not sure regarding its purpose on the project. However, implementing the BL on the use-case will help us to be more flexible in the future! Thank you for bringing it up, ill update the PR.